gems/pending/disk/modules/miq_disk_cache.rb

```diff
@@ -105,11 +120,16 @@ def hit_or_miss
   end
 
   def entry_range(start_sector, number_sectors)
-    real_start_block, sector_offset = start_sector.divmod(@min_sectors_per_entry)
-    number_blocks     = number_sectors % @min_sectors_per_entry
-    sectors_to_read   = (number_blocks + (sector_offset > 0 ? 1 : 0)) * @min_sectors_per_entry
-    real_start_sector = real_start_block * @min_sectors_per_entry
-    end_sector        = real_start_sector + sectors_to_read - 1
+    # Cache entries are *multiples* of MIN_SECTORS_PER_ENTRY * @blocksize  in length,
+    # aligned to MIN_SECTORS_PER_ENTRY * @blocksize byte boundaries.
+    # real_start_block is the aligned cache block based on the start_sector, and
+    # real_start_sector is the disk sector for that cache block.
+    real_start_block    = start_sector / MIN_SECTORS_PER_ENTRY
+    real_end_block      = (start_sector + number_sectors) / MIN_SECTORS_PER_ENTRY
+    number_cache_blocks = real_end_block - real_start_block + 1
+    sectors_to_read     = number_cache_blocks * MIN_SECTORS_PER_ENTRY
+    real_start_sector   = real_start_block * MIN_SECTORS_PER_ENTRY
```

Comment:
You've broken the ability to configure these parameters. In this method, you need to use the instance variables instead of the constants.